packages/components/view-switcher/src/view-switcher.stories.tsx

```diff
@@ -79,67 +63,65 @@ export const Default: Story = {
     iconOnlyButton4: categorize(BASE_CATEGORY, 4),
     isDisabledButton4: categorize(BASE_CATEGORY, 4),
   },
-  render: (args) => (
-    <ViewSwitcher.Group {...args}>
-      {[...Array(BUTTONS_COUNT).keys()].map((index) => {
-        const i = index + 1;
+} satisfies Meta<ViewSwitcherPropsAndCustomArgs>;
+
+export default meta;
+
+type Story = StoryObj<ViewSwitcherPropsAndCustomArgs>;
+
+const ViewSwitcherStoryWrapper = (props: ViewSwitcherPropsAndCustomArgs) => (
+  <ViewSwitcher.Group {...props}>
+    {[...Array(BUTTONS_COUNT).keys()].map((index) => {
+      const i = index + 1;
+      return (
+        <ViewSwitcher.Button
+          key={i}
+          label={`View ${i}`}
+          isDisabled={props[`isDisabledButton${i}`]}
+          value={`Button ${i}`}
+          icon={props[`iconViewSwitcherButton${i}`] as unknown as ReactElement}
```

Comment:
This is the only cast we need to keep as Storybook is converting the initial `string` to an actual `Icon` element under the hood.